World.js

```diff
@@ -7,7 +7,14 @@ function World () {
 }
 
 World.prototype.tour = function () {
-  return Object.keys(this.playgrounds);
+  var tour = Object.keys(this.playgrounds);
+  var i = tour.length;
+  while (i--) {
+    if (this.playgrounds[tour[i]].isEmpty()) {
+      tour.splice(i, 1);
```

Comment:
Whoa there! If that's not a perfect use case for `Array.prototype.filter` I don't know what is 😄

Reply:
Programming is so hard ! I can't manage to get it all in my brain.